jwst/msaflagopen/msaflag_open.py

```diff
@@ -0,0 +1,235 @@
+from __future__ import division
+
+#
+#  Module for flagging the DQ array of pixels affected by failed
+#  open MSA shutters in nirspec science data sets
+#
+import json
+import math
+import numpy as np
+import logging
+from .. import datamodels
+from ..assign_wcs.nirspec import slitlets_wcs, nrs_wcs_set_input
+from ..transforms.models import Slit
+from gwcs.wcs import WCS
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+
+FAILEDOPENFLAG = datamodels.dqflags.pixel['MSA_FAILED_OPEN']
+SHUTTERS_PER_ROW = 365
+#
+# States in the msaoper file that are to flagged when set to 'open'
+FLAGGABLE_STATES = ['Internal state', 'TA state', 'state']
+
+def do_correction(input_model, shutter_refname, wcs_refnames):
+    """
+    Short Summary
+    -------------
+    Apply DQ flag to pixels affected by failed open MSA shutters
+
+    Parameters
+    ----------
+    input_model: datamodel object
+        science data to be corrected
+
+    shutter_refname: string
+        Name of MSAOPER reference file
+
+    wcs_refnames: dict
+        Dictionary of wcs reference file names
+
+    Returns
+    -------
+    output_model: data model object
+        science data with DQ array modified
+
+    """
+    #
+    # Create a list of failed open slitlets from the msaoper reference file
+    failed_slitlets = create_slitlets(input_model, shutter_refname)
+
+    # Flag the stuck open hutters
+    output_model = flag(input_model, failed_slitlets, wcs_refnames)
+
+    output_model.meta.cal_step.msaflagopen = 'COMPLETE'
```

Comment:
Right now the attribute name for this in core.schema.yaml is "msa_flagging", as opposed to "msaflagopen." So one or the other will need to be updated to make them match.